packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -650,8 +663,25 @@ class PaddleSliderValueIndicatorShape extends SliderComponentShape {
 
     // This is the needed extra width for the label.  It is only positive when
     // the label exceeds the minimum size contained by the round top lobe.
-    final double halfWidthNeeded =
-        math.max(0.0, inverseTextScale * labelHalfWidth - (_topLobeRadius - _labelPadding));
+    final double halfWidthNeeded = math.max(
+      0.0,
+      inverseTextScale * labelHalfWidth - (_topLobeRadius - _labelPadding),
+    );
+
+    double shift = _getIdealOffset(parentBox, halfWidthNeeded, scale * textScaleFactor, center);
+    double leftWidthNeeded;
+    double rightWidthNeeded;
+    if (shift < 0.0) {
```

Comment:
It looks like this could be simplified a little:
```
shift = math.max(shift, shift.sign * halfWidthNeeded);
rightWidthNeeded = halfWidthNeeded + shift;
leftWidthNeeded = halfWidthNeeded - shift;
```

Reply:
No, because one needs to be min and one needs to be max.  I did pull out the last two assignments, though, so it's simpler.